File: data-visualization/gists/quarterly-profit-barplots.R
labels = c("2008 Q1", "Q2", "Q3", "Q4",
            "2009 Q1", "Q2", "Q3", "Q4")
profit = c(45, 32, 67, 23, 42, 56, 64, 92)+40

# Draw a bar-plot
bp <- barplot(profit,
                names.arg=labels,
                axisnames=TRUE,
                ylab="Quarterly profit ($ '000)",
                border = TRUE)
text(bp, profit+3,
        labels=format(profit),
        xpd = TRUE,
        col = "black")

# Now rather use a line plot.
# Graph profit, but turn off axes
# and annotations
plot(profit, type="b", axes=TRUE,
        ann=FALSE, xaxt="n")

# Show the x-axis using our labels
axis(1, at=1:8, lab=labels)

# Plot title
title(ylab="Quarterly profit ($ '000)")

File: data-visualization/gists/quarterly-profit-barplots.py
import pandas as pd

pd.options.plotting.backend = "plotly"

labels = ["2008 Q1", "Q2", "Q3", "Q4", "2009 Q1", "Q2", "Q3", "Q4"]
profit = pd.DataFrame(
    data=[45, 32, 67, 23, 42, 56, 64, 92],
    index=labels,
    columns=["Quarterly profit ($ '000)"],
) + 40

# Draw a bar plot.
fig = profit.plot.bar()
fig.update_traces(
    text=profit["Quarterly profit ($ '000)"],
    textposition="outside",
)
fig.update_layout(
    yaxis_title_text="Quarterly profit ($ '000)",
    showlegend=False,
)
fig.show()

# Now rather use a line plot.
fig = profit.plot.line(markers=True)
fig.update_layout(
    yaxis_title_text="Quarterly profit ($ '000)",
    showlegend=False,
)
fig.show()
